src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp

```diff
@@ -95,71 +121,45 @@ struct boss_murmurAI : public CombatAI
         if (m_creature->IsInCombat())
             return;
 
-        // kill one that's moving
-        if (urand(0, 1))
+        // First cast after door opens should always be a Suprression Blast
+        if (!m_firstCast)
         {
-            GuidVector moversVector;
-            for (ObjectGuid& guid : spellbindersVector)
-            {
-                if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
-                {
-                    if (creature->GetMotionMaster()->GetCurrentMovementGeneratorType() == WAYPOINT_MOTION_TYPE)
-                    {
-                        moversVector.push_back(guid);
-                    }
-                }
-            }
-            for (ObjectGuid& guid : summonersVector)
+            DoCastSpellIfCan(nullptr, SPELL_SUPPRESSION_BLAST);
+            m_firstCast = true;
+        }
+        else
+        { 
+            // Kill a moving target
+            if (urand(0, 1))
             {
-                if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
+                GuidVector m_WrathTargetGuid;
+                std::vector<Creature*> const* m_WrathTarget = m_creature->GetMap()->GetCreatures(MURMURS_WRATH_TARGETS_03);
+                if (m_WrathTarget)
                 {
-                    if (creature->GetMotionMaster()->GetCurrentMovementGeneratorType() == WAYPOINT_MOTION_TYPE)
-                    {
-                        moversVector.push_back(guid);
-                    }
+                    for (Creature* creature : *m_WrathTarget)
+                        if(creature->IsAlive() && !creature->IsInCombat())
+                            m_WrathTargetGuid.push_back(creature->GetObjectGuid());
                 }
-            }
-            if (moversVector.size() > 0)
-            {
-                if (ObjectGuid& guid = moversVector[urand(0, moversVector.size() - 1)])
+
+                if (m_WrathTargetGuid.size() > 0)
                 {
-                    if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
+                    if (ObjectGuid& guid = m_WrathTargetGuid[urand(0, m_WrathTargetGuid.size() - 1)])
```

Comment:
Useless to have this in an if and also useless to use &. ObjectGuid is 64-bit and a ref is 64-bit. Would only make sense if you wanted to edit it after.

Reply:
should be fixed